launcher/ui/instanceview/InstanceView.cpp

```diff
@@ -925,16 +926,23 @@ QModelIndex InstanceView::moveCursor(QAbstractItemView::CursorAction cursorActio
             if (column > 0) {
                 m_currentCursorColumn = column - 1;
                 return cat->rows[row][column - 1];
+            } else if (row > 0) {
+                row -= 1;
+                int newRowSize = cat->rows[row].size();
+                m_currentCursorColumn = newRowSize - 1;
+                return cat->rows[row][m_currentCursorColumn];
```

Comment:
Seems as I've tested by your description it does all those functions,  but since I'm at job can you confirm? Thanks.

Reply:
Right now(using this pull request latest commit) the keys left and right move only over the current group instances.
I want them to also move between groups(this doesn't happen right now):
>If is already the first element from group, and left is pressed, try to move to the last element from the last group.
And if is the last element, and right is pressed, try to move to first from next group.

Reply:
> Right now(using this pull request latest commit) the keys left and right move only over the current group instances. I want them to also move between groups(this doesn't happen right now):
> 
> > If is already the first element from group, and left is pressed, try to move to the last element from the last group.
> > And if is the last element, and right is pressed, try to move to first from next group.

Hmm it moved also between groups with arrow keys if the there is last instance on group then it switches to next group with end button it will switch to last instance on group and by pressing home button it swaps back to first instance on group

Reply:
If I use only left and right arrow keys this doesn't work for me.
This also can be observed from the code change: you iterate over the current group only.

Reply:
> If I use only left and right arrow keys this doesn't work for me. This also can be observed from the code change: you iterate over the current group only.

Done

Reply:
I might improve whole function with helpers instead